File: app/src/main/java/com/enipro/data/remote/model/FeedContent.java
package com.enipro.data.remote.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.graphics.Bitmap;
import android.os.Parcel;
import android.os.Parcelable;

import com.enipro.Application;
import com.enipro.db.EniproDatabase;
import com.google.firebase.database.Exclude;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

@org.parceler.Parcel
@Entity(tableName = "feed_content")
public class FeedContent {

    @PrimaryKey
    public int content_id = new Random().nextInt();

    @SerializedName("text")
    @Expose
    public String text;

    @SerializedName("image")
    @Expose
    public String image;

    @SerializedName("video")
    @Expose
    public String video;

    @SerializedName("doc")
    @Expose
    public Document doc;

    public boolean docExists = false;

    /**
     * Type of media item for this feed content.
     */
    public int mediaType;

    public FeedContent() {
    }


    public int getMediaType() {
        return this.mediaType;
    }

    public void setMediaType(int mediaType) {
        this.mediaType = mediaType;
    }

    public void setContent_id(int content_id) {
        this.content_id = content_id;
    }

    public int getContent_id() {
        return this.content_id;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getText() {
        return this.text;
    }

    public void setVideo(String video) {
        this.video = video;
    }

    public String getVideo() {
        return this.video;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getImage() {
        return this.image;
    }

    /**
     * Returns a name used to save the comment image file.
     *
     * @return
     */
    public String getMediaName() {
        return Application.getActiveUser().get_id().get_$oid() + new java.util.Random().nextLong() + new java.util.Date();
    }

    public Document getDoc() {
        return doc;
    }

    public void setDoc(Document doc) {
        this.doc = doc;
    }

    public boolean isDocExists() {
        return docExists;
    }

    public void setDocExists(boolean docExists) {
        this.docExists = docExists;
    }

    public class MediaType {
        public static final int IMAGE = 0x005;
        public static final int VIDEO = 0x002;
        public static final int DOC = 0x003;
    }
}

File: app/src/main/java/com/enipro/data/remote/model/FeedContent.kt
package com.enipro.data.remote.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import android.os.Parcel
import android.os.Parcelable
import com.enipro.Application
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

@Entity(tableName = "feed_content")
class FeedContent() : Parcelable {

    @PrimaryKey
    var content_id = Random().nextInt()

    @SerializedName("text")
    @Expose
    var text: String? = null

    @SerializedName("image")
    @Expose
    var image: String? = null

    @SerializedName("video")
    @Expose
    var video: String? = null

    @SerializedName("doc")
    @Expose
    var doc: Document? = null

    var isDocExists = false
    var mediaType: Int = 0

    val mediaName: String
        get() = Application.getActiveUser().id + java.util.Random().nextLong() + java.util.Date()

    constructor(parcel: Parcel) : this() {
        content_id = parcel.readInt()
        text = parcel.readString()
        image = parcel.readString()
        video = parcel.readString()
        doc = parcel.readParcelable(Document::class.java.classLoader)
        isDocExists = parcel.readByte() != 0.toByte()
        mediaType = parcel.readInt()
    }

    object MediaType {
        val IMAGE = 0x005
        val VIDEO = 0x002
        val DOC = 0x003
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(content_id)
        parcel.writeString(text)
        parcel.writeString(image)
        parcel.writeString(video)
        parcel.writeParcelable(doc, flags)
        parcel.writeByte(if (isDocExists) 1 else 0)
        parcel.writeInt(mediaType)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FeedContent> {
        override fun createFromParcel(parcel: Parcel): FeedContent {
            return FeedContent(parcel)
        }

        override fun newArray(size: Int): Array<FeedContent?> {
            return arrayOfNulls(size)
        }
    }
}
